lib/helpers/createServerMiddleware.js

```diff
@@ -1,20 +1,77 @@
 const admin = require('firebase-admin')
 const logger = require('../logger')
 
+const simulateUserRecord = ({
```

Comment:
Won't have much time tonight to look at it in more detail, but is there a way to avoid this?

This will make the maintenance of this module very cumbersome. When Firebase changes just a small thing here, we will always have to rush and change the same in the module, too.

Furthermore it makes it a pain for anyone who wants to contribute. The more code, the less people will wanna look at it.

If possible I would like to avoid something like this...

What's your opinion? It might even be a good reason to just drop the entire "no-credential" idea.

hmm..

Reply:
:thinking: 
I see your point...
I just did this to provide a stable API to the user, but your quite right.

What about we just guarantee `{ uid, email, emailVerified, displayName }` and put all additional claims into `{ /* ... */, customClaims }`?

Reply:
Hmm yeah... 🤔

I see that it would make sense to provide a stable API and return the same object, but on the other hand this module should also be stable in the future without us having to intervene after every Firebase upgrade.

> What about we just guarantee { uid, email, emailVerified, displayName }
Maybe that's a good trade-off. And if someone needs all data there is always the option to provide credentials. 
> and put all additional claims into { /* ... */, customClaims }?
You mean you want to mix it with custom claims?

Reply:
> You mean you want to mix it with custom claims?

`admin.auth.DecodedIdToken` is just the decoded JWT Token with an added `uid` property, custom claims live on the root of this object as they do on the object returned by `firebase.auth().currentUser.getIdTokenResult()` in that sense it might even be desirable as it connects better to the `onAuthStateChangedMutation/Action`

Reply:
> > You mean you want to mix it with custom claims?
> 
> `admin.auth.DecodedIdToken` is just the decoded JWT Token with an added `uid` property, custom claims live on the root of this object as they do on the object returned by `firebase.auth().currentUser.getIdTokenResult()` in that sense it might even be desirable as it connects better to the `onAuthStateChangedMutation/Action`

Oh, right, I forgot. Alright, then let's do it that way :) 👍

Reply:
I have opted to create a custom user object with the following specification:

```d.ts
export type FireAuthServerUser = Omit<auth.UserRecord, 'disabled' | 'metadata' | 'providerData'>
  & Partial<Pick<auth.UserRecord, 'disabled' | 'metadata' | 'providerData'>>
  & { allClaims: auth.DecodedIdToken }
```
Essentially we provide a "light" version of the `auth.UserRecord` without `credential` and augment to full if `credential` is set.
The `admin.auth.DecodedIdToken` is set on a separate property (`allClaims`) so it remains consistent across both implementations.

Let me know what you think.